### DISCUSSION TITLE
How to build a dll file for CppAD?

### DISCUSSION THREAD

**Discussion by Wang-Peng-debug** [coin-or/CppAD#215] [Q&A]
The CppAD running discontinued for some strange errors, which I thought is caused by wrong dynamical connection. So I need the dll file of CppAD. Does anyone know how to build it?

**Comment by bradbell** [comment]
How are you building and checking CppAD; i.e., 

1. Are you using Minsys or a Visual Studio command window ?
2. Were there any errors reported by your cmake comamnd ?
3. Were there any errors reported by your make check command ?

**Reply by Wang-Peng-debug** [reply]
![image](https://github.com/user-attachments/assets/569fb072-6cf2-4f67-830e-fb9e9408692e)
I use Cmake as above, and I have never heard about Minsys. Due to the invalid dynamic link, some grammar errors occurred while the program was running.

**Comment by bradbell** [comment]
There are many versions of cmake under windows, some are for minsys and one is for Visual Studio. You will get the one for visual studio if you use a Visual Studio command window. I like only use the command interface to cmake and not the gui so that I know exactly what has been set by the script that runs the cmake command.

**Reply by Wang-Peng-debug** [reply]
Ok, thank you anyway, now I think this problme has been solved, but then I still need to solve the next one, LOL!!!!!!!
